### PR TITLE
Test power-bar priority rules + example docs (#48 #49)

### PR DESCRIPTION
**Pull request by jasonherald**
## Summary

Wave 7 test-coverage pass — two test/doc-themed improvements bundled.

### #48 — Refactor + test `auto_detect_power_bar` priority rules

`detect_lock` and `detect_command` had the same "skip if non-empty, iterate candidates, take first that passes a probe, store" shape. Extracted into `pick_first_present(slot, candidates, probe)` — pure function, probe-injected, fully testable without depending on real `PATH` state.

The "explicit flag wins, slot only fills if empty" rule from the `--pb-auto` README contract is now enforced in one place. A regression where `detect_command` silently overwrote a pre-set slot would have to fail the `pick_first_present_preserves_prefilled_slot` test.

Four pure-function tests covering the action plan acceptance:
- pre-filled slot preserved (no probe call, no clobber)
- first match wins (probe rejects head, accepts mid)
- no match → slot stays empty
- empty candidate list → slot stays empty (defensive)

### #49 — Example doc blocks for `subsequence_match` and `truncate`

Added `# Examples` sections covering the canonical call patterns:

- `subsequence_match`: subsequence vs out-of-order, case-insensitivity, empty-needle behavior.
- `truncate`: short-input passthrough, long-ASCII ellipsis, **the char-vs-byte gotcha** (CJK input where byte-counting would give surprising results).

**Tagged `rust,ignore`** because this is a binary-only crate — `cargo test --doc` errors with "no library targets found in package nwg-drawer", so doctests can't actually run. The action plan acceptance asked for `cargo test` to exercise the doctests; that would require introducing a `lib.rs` target purely for doctest plumbing, which is more scaffolding than the value justifies.

The blocks render as syntax-highlighted Rust in `cargo doc` and serve as documentation only; runtime regression coverage stays in the existing `#[cfg(test)] mod tests` blocks.

## Test plan

- [x] `make lint` — fmt + clippy + test + deny + audit clean
- [x] `cargo test` — 102 tests pass (was 98; +4 for `pick_first_present`)
- [x] `cargo doc` would render the new `# Examples` sections (not verified but standard rustdoc behavior)

## Summary by CodeRabbit

* **Refactor**
  * Improved internal detection logic to preserve prefilled values and ensure predictable first-match selection behavior.

* **Documentation**
  * Expanded usage examples and explanatory notes for sequence-matching and truncation utilities to clarify behavior and edge cases.

* **Tests**
  * Added unit tests covering prefilling preservation, first-match selection, no-match behavior, and empty-candidate scenarios.

### CODE REVIEW

File: src/power_bar_detect.rs [hunk 1]
```diff
@@ -80,3 +94,44 @@ fn can_suspend() -> bool {
...
+mod tests {
+    use super::*;
+
+    #[test]
+    fn pick_first_present_preserves_prefilled_slot() {
+        // Explicit user value must never be clobbered, even if every
+        // candidate would otherwise match.
+        let mut slot = "user-set-cmd".to_string();
+        pick_first_present(&mut slot, &["a", "b"], |_| true);
+        assert_eq!(slot, "user-set-cmd");
```
[the remark this answers is not in the document]
Comment (jasonherald) [reply src/power_bar_detect.rs:113]: Good catch — fixed in 5fb847f. Probe now `panic!`s if invoked, so the test fails on any path that walks candidates before checking `slot.is_empty()`. The original assertion-only test would have passed even if the probe ran (probe returns `true` → first candidate matches the slot’s existing value `"user-set-cmd"` only by coincidence, no visible diff).
